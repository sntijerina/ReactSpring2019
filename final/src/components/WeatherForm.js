import React, { useState } from "react";
import Button from '@material-ui/core/Button';
import TextField from '@material-ui/core/TextField';

export default function WeatherForm(props) {
    const [city, setCity] = useState(undefined);

    function handleSubmit (event) {
        event.preventDefault();
        typeof props.onFormSubmit === 'function' &&
            props.onFormSubmit(city);
    }

    return (
        <form onSubmit = {handleSubmit}>
                <TextField 
                    type="text" 
                    name="city" 
                    placeholder="City..." 
                    onChange={event => setCity(event.target.value)}
                />
                <Button  
                    variant="contained" 
                    color="primary" 
                    type="submit">
                    Get Weather
                </Button>
        </form>
    )
}

