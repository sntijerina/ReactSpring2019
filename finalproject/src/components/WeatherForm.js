import React, { Component } from "react";
import PropTypes from "prop-types";
import Grid from "@material-ui/core/Grid";
import Card from "@material-ui/core/Card";

class WeatherForm extends Component {
    constructor() {
        super();
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleSubmit(event) {
        event.preventDefault();
        
        typeof this.props.onFormSubmit === 'function' &&
            this.props.onFormSubmit(event.target.elements.city.value);
    }

    render () {
        return (
            <form onSubmit = {this.handleSubmit}>
                    <input type="text" name="city" placeholder="City..."/>
                    <button type="submit">Get Weather</button>
            </form>
        );
    }
}

export default WeatherForm;
