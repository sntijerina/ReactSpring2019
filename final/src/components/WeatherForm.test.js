import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import WeatherForm from './WeatherForm';
import Adapter from "enzyme-adapter-react-16";
import { TextField, Button } from '@material-ui/core';

Enzyme.configure({ adapter: new Adapter() });

describe('Weather Form', () => {

    it('Weather Form renders', () => {
        const component = shallow(<WeatherForm />);
    
        expect(component).toMatchSnapshot();
    });

    it('City renders', () => {
        const component = shallow(<WeatherForm />);

        expect(component.find(TextField).length).toEqual(1);
    });

    it('Button renders', () => {
        const component = shallow(<WeatherForm />);
        
        expect(component.find(Button).length).toEqual(1);
    });
});