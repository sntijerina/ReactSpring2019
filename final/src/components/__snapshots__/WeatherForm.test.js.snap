// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather Form Weather Form renders 1`] = `
<form
  onSubmit={[Function]}
>
  <TextField
    name="city"
    onChange={[Function]}
    placeholder="City..."
    required={false}
    select={false}
    type="text"
    variant="standard"
  />
  <WithStyles(Button)
    color="primary"
    type="submit"
    variant="contained"
  >
    Get Weather
  </WithStyles(Button)>
</form>
`;
